src/posix/sdl/i_input.cpp

```diff
@@ -306,55 +306,46 @@ void MessagePump (const SDL_Event &sev)
 
 	case SDL_MOUSEBUTTONDOWN:
 	case SDL_MOUSEBUTTONUP:
-	case SDL_MOUSEMOTION:
-		if (!GUICapture || (sev.type != SDL_MOUSEMOTION && (sev.button.button == 4 || sev.button.button == 5)))
+		if (!GUICapture)
 		{
-			if(sev.type != SDL_MOUSEMOTION)
+			/* [GUT] The code that was here before was only tested on Gentoo with a singular mouse
+			 * and made some assumptions about SDL that you aren't supposed to make, leading to undefined behaviour
+			 * This has been tested by multiple people who run Linux so it should be far more reliable
+			*/
+			event.type = sev.type == SDL_MOUSEBUTTONDOWN ? EV_KeyDown : EV_KeyUp;
+			switch (sev.button.button)
 			{
-				event.type = sev.type == SDL_MOUSEBUTTONDOWN ? EV_KeyDown : EV_KeyUp;
-				/* These button mappings work with my Gentoo system using the
-				* evdev driver and a Logitech MX510 mouse. Whether or not they
-				* carry over to other Linux systems, I have no idea, but I sure
-				* hope so. (Though buttons 11 and 12 are kind of useless, since
-				* they also trigger buttons 4 and 5.)
-				*/
-				switch (sev.button.button)
-				{
-				case SDL_BUTTON_LEFT:	event.data1 = KEY_MOUSE1;		break;
-				case SDL_BUTTON_MIDDLE:	event.data1 = KEY_MOUSE3;		break;
-				case SDL_BUTTON_RIGHT:	event.data1 = KEY_MOUSE2;		break;
-				case 8:		event.data1 = KEY_MOUSE4;		break; // For whatever reason my side mouse buttons are here.
-				case 9:		event.data1 = KEY_MOUSE5;		break;
-				case SDL_BUTTON_X1:		event.data1 = KEY_MOUSE6;		break; // And these don't exist
-				case SDL_BUTTON_X2:		event.data1 = KEY_MOUSE7;		break;
-				case 6:		event.data1 = KEY_MOUSE8;		break;
-				default:	printf("SDL mouse button %s %d\n",
-					sev.type == SDL_MOUSEBUTTONDOWN ? "down" : "up", sev.button.button);	break;
-				}
-				if (event.data1 != 0)
-				{
-					D_PostEvent(&event);
-				}
+			case SDL_BUTTON_LEFT:	event.data1 = KEY_MOUSE1;		break;
+			case SDL_BUTTON_MIDDLE:	event.data1 = KEY_MOUSE3;		break;
+			case SDL_BUTTON_RIGHT:	event.data1 = KEY_MOUSE2;		break;
+			case SDL_BUTTON_X1:		event.data1 = KEY_MOUSE4;		break;
+			case SDL_BUTTON_X2:		event.data1 = KEY_MOUSE5;		break;
+			case 6:		event.data1 = KEY_MOUSE6;		break;
+			case 7:		event.data1 = KEY_MOUSE7;		break;
+			case 8:		event.data1 = KEY_MOUSE8;		break;
+			default:	printf("SDL mouse button %s %d\n",
+				sev.type == SDL_MOUSEBUTTONDOWN ? "down" : "up", sev.button.button);	break;
+			}
+			if (event.data1 != 0)
+			{
+				D_PostEvent(&event);
 			}
 		}
-		else if (sev.type == SDL_MOUSEMOTION || (sev.button.button >= 1 && sev.button.button <= 3))
+		else if ((sev.button.button >= SDL_BUTTON_LEFT && sev.button.button <= SDL_BUTTON_X2))
 		{
 			int x, y;
 			SDL_GetMouseState (&x, &y);
 
 			event.data1 = x;
 			event.data2 = y;
 
-			screen->ScaleCoordsFromWindow(event.data1, event.data2);
```

Comment:
Is this line removed intentionally?

Reply:
No, it isn't. I'll add this back in, it must have got removed in the reorganising.